src/actions/guides/json_and_apis/cors.cr

```diff
@@ -21,12 +23,167 @@ class Guides::JsonAndApis::Cors < GuideAction
         response.headers["Access-Control-Allow-Origin"] = "*"
         response.headers["Access-Control-Allow-Headers"] =
           "DNT,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Range"
+        response.headers["Access-Control-Allow-Methods"] = "*"
         continue
       end
     end
     ```
 
-    > Note: the before action method must return `continue`, or a `LuckyWeb::Response`. See [Actions and Routing](#{Guides::HttpAndRouting::RequestAndResponse.path(anchor: Guides::HttpAndRouting::RequestAndResponse::ANCHOR_RUN_CODE_BEFORE_OR_AFTER_ACTIONS_WITH_PIPES)}) for more info.
+    This will set these headers before every action in your API.
+
+    > Note: the `before` pipe must return `continue`, or a `LuckyWeb::Response`. See [Actions and Routing](#{Guides::HttpAndRouting::RequestAndResponse.path(anchor: Guides::HttpAndRouting::RequestAndResponse::ANCHOR_RUN_CODE_BEFORE_OR_AFTER_ACTIONS_WITH_PIPES)}) for more info.
+
+    ### Complex example
+
+    You may find that you need a little more than a simple setup to handle CORS.
+    Some requests may require a preflight `OPTIONS` call. We have two ways we can handle this.
```

Comment:
I think we can remove the part about `match :options` since it is so inflexible. I bet the handler is the right call almost every time